### ISSUE TITLE
Slow virtual machines in VMM

### ISSUE THREAD

**Issue by upais** [AuxXxilium/arc#179]
There is a problem with slow performance of virtual machines through VMM. It is relevant for ARPL and TCRP Friend, while it does not exist on TCRP.

More details:
https://github.com/fbelavenuto/arpl/issues/260
https://github.com/pocopico/tinycore-redpill/issues/208

I have tested on ARPL v1.1-beta2a and also tried the ARC version from a week ago - the problem is relevant. One person on the forum appears to have figured out the issue, but I cannot confirm it yet as the changes involved prevent my DSM from starting up. Link: https://xpenology.com/forum/topic/53817-tinycore-redpill-loader-tcrp/?do=findComment&comment=442030

Could you please check if it is possible to fix this problem in ARC?

**Comment by AuxXxilium** [comment]
https://github.com/fbelavenuto/arpl/issues/260#issuecomment-1381702401

since we use same base, this issue shouldn't appear again.

**Comment by upais** [comment]
There were two problems there, and the resolved one was due to CPU performance:

https://xpenology.com/forum/topic/65408-automated-redpill-loader-arpl/?do=findComment&comment=422812

**Comment by AuxXxilium** [comment]
remove powersched from addons and try again

**Comment by upais** [comment]
I will try tomorrow, but this issue has been present since powersched was not in ARPL.

**Comment by AuxXxilium** [comment]
it was fixed. and i have no issue so far.

**Comment by AuxXxilium** [comment]
btw: https://browser.geekbench.com/v6/cpu/1005404

Benchmark from my nas. So CPU is not the problem.

**Comment by upais** [comment]
So it's not about CPU performance

**Comment by AuxXxilium** [comment]
Do you use this SHR trash raid?

**Comment by upais** [comment]
I tested on raid1 (2 hdd) or 1 hdd

**Comment by AuxXxilium** [comment]
I installed a new Windows VM in under 6 minutes. So no VMM issue for me.

**Comment by AuxXxilium** [comment]
do you have rechecked it with latest versions?

**Comment by upais** [comment]
Yesterday and today I tried 23.5.1 and 23.4.3 several times, but DSM appears on the network for 10 seconds and then disappears. I tried Legacy and UEFI mode, and built a new loader around five times. I also tried the latest changes to TCRP Friend in M-Shell mode, but the problem persists. I believe this may be a specific issue with my configuration, although it is unclear why TCRP (no Friend mode) does not have this problem.

If you have any ideas why this might be happening, I would be happy to test them. I don't see the point of "powersched." Later, I will try to install something other than DVA3221, maybe that could be the issue.

**Comment by AuxXxilium** [comment]
Please, send me a Screenshot from Sysinfo Menu in Arc Loader.

**Comment by upais** [comment]
After or before building the bootloader?

**Comment by AuxXxilium** [comment]
after

**Comment by upais** [comment]
Okay, I will do it tomorrow. If you have any ideas, please let me know. I will test several options and send you the data.

**Comment by AuxXxilium** [comment]
Screenshot first, than we can look for the issue.

**Comment by upais** [comment]
New bootloader, clean HDD: 5-10 seconds and DSM disappears from the network, waited for 5 minutes. Screenshots:
![IMG_0240](https://user-images.githubusercontent.com/86227709/235514388-2863a02e-962e-41d6-86b7-c5fc7b191bce.JPG)
![IMG_0241](https://user-images.githubusercontent.com/86227709/235514396-fdce0020-52ec-4d6c-a4f1-17afc356f1f8.JPG)
![IMG_0242](https://user-images.githubusercontent.com/86227709/235514400-8d6a5d40-056b-4bf2-9b70-f9b9bb62b8d2.JPG)

**Comment by AuxXxilium** [comment]
first you have an issue, you try to use 6 sata ports on an 4 bay nas. second, your disk is not in slot 1 of your controller. so you have to use sata remap or a DT model.

**Comment by upais** [comment]
Strange, I've already built several dozens of bootloaders in this configuration, including ARC. This is my testing configuration.

**Comment by AuxXxilium** [comment]
can't work. best you use a dt model. like 2422+ or 1621xs+

**Comment by AuxXxilium** [comment]
the next point is, it looks like is dedecting you sata is in raid mode and not ahci mode.

**Comment by upais** [comment]
I always use 'satamap=6' without fail. Initially, in TCRP, it was necessary to fill in these attributes manually, but then the function "./rploader.sh satamap" appeared, automatically filling them in, and I always set it to "6". The same goes for ARPL.

AHCI mode is enabled, I checked it today when resetting BIOS settings and testing bootloaders.

**Comment by AuxXxilium** [comment]
i told you 2 ways. i can't do anything if other loader have bugs. thats why i added these options, to make it clear and no one have to do something that can go wrong. we all use redpill as base, but no other dev has fixed so many bugs in redpill.

use remap and try again or use a dt model, that doesn't need this setting.

**Comment by upais** [comment]
Did I understand correctly that the bootloader limits the number of disks according to the original model? In other words, you cannot insert more than 4 disks into the DVA3221?

**Comment by AuxXxilium** [comment]
correct, but you can, some models allow it, some not. models with extension support, allow more disks. the hardlimit of dsm is 24. i tried with rs4021xs+.

**Comment by upais** [comment]
Questionable fix, but it's your call. Also, ARC was working with satamap=6 a week ago when I was testing the issue in VMM.

I understand the essence, but this bootloader doesn't suit me. Thank you.

**Comment by AuxXxilium** [comment]
than select remap after selecting model and try again.

**Comment by upais** [comment]
Fulfilled the recommendations, DSM started.
Virtual machines work slowly, as described earlier.
Screenshot from Sysinfo.
![IMG_0255](https://github.com/AuxXxilium/arc/assets/86227709/a45fdacf-4531-4f90-b32d-4397e473b42d)
